pandas/tests/frame/test_query_eval.py

```diff
@@ -160,144 +165,175 @@ def test_eval_resolvers_as_list(self):
 
 class TestDataFrameQueryWithMultiIndex(object):
 
-    def test_query_with_named_multiindex(self, parser, engine):
+    def test_query_with_named_multiindex(self, parser, engine,
+                                         partial_str_match):
         skip_if_no_pandas_parser(parser)
-        a = np.random.choice(['red', 'green'], size=10)
-        b = np.random.choice(['eggs', 'ham'], size=10)
+        a = np.random.choice(['red', 'a_red', 'a_red_a',
+                              'red_a', 'Red', 'green'], size=30)
+        b = np.random.choice(['eggs', 'ham'], size=30)
         index = MultiIndex.from_arrays([a, b], names=['color', 'food'])
-        df = DataFrame(np.random.randn(10, 2), index=index)
+        df = DataFrame(np.random.randn(30, 2), index=index)
         ind = Series(df.index.get_level_values('color').values, index=index,
                      name='color')
+        kwargs = {'parser': parser, 'engine': engine,
+                  'partial_str_match': partial_str_match}
 
         # equality
-        res1 = df.query('color == "red"', parser=parser, engine=engine)
-        res2 = df.query('"red" == color', parser=parser, engine=engine)
+        res1 = df.query('color == "red"', **kwargs)
```

Comment:
Can you just be explicit about the parameter being passed instead of using kwargs?